In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("/Users/roman2006/PycharmProjects/Liana Clone/Session_2/2_titanic_dataset.csv")

In [3]:
df.shape

(891, 12)

In [4]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

###  1. How many people died?

In [7]:
df["Survived"].value_counts()

0    549
1    342
Name: Survived, dtype: int64

In [8]:
549/(549+342)

0.6161616161616161

### Knowledge found: 549 people died on the Titanic, a total of 61.6% of the passengers.

### 2. How many people died in first class?

In [52]:
sum([(df['Survived']==0) &(df['Pclass']==1)]).sum()

80

In [58]:
80/549

0.14571948998178508

### Knowledge found: 80 people died from first class, 14.6% of the people who died

### 3. How many people died in second class?

In [60]:
sum([(df['Survived']==0) &(df['Pclass']==2)]).sum()

97

In [61]:
97/549

0.1766848816029144

### Knowledge found: 97 people died from second class, 17.7% of the people who died

### 4. How many people died in third class?

In [63]:
sum([(df['Survived']==0) &(df['Pclass']==3)]).sum()

372

In [64]:
372/549

0.6775956284153005

### Knowledge found: 372 people died from third class, 67.75% of the people who died

### 5. What was the survivability of first class?

In [68]:
1-80/sum(df['Pclass']==1)

0.62962962962962965

### Knowledge found: The survivability rate of first class was of 63%

### 6. What was the survivability of second class?

In [69]:
1-97/sum(df['Pclass']==2)

0.47282608695652173

### Knowledge found: The survivability rate of first class was of 47.3%

### 7. Whatwas the survivability of third class?

In [71]:
1-372/sum(df['Pclass']==3)

0.24236252545824843

### Knowledge found: The survivability rate of first class was of 24.2%

### 8. What was the overall survibality?

In [73]:
1-549/(549+342)

0.38383838383838387

### Knowledge found: The overall survivability rate was of 38.4%

### 9. Gender rate of the people aboard the Titanic?

In [77]:
df["Sex"].value_counts()

male      577
female    314
Name: Sex, dtype: int64

In [78]:
577/(577+314)

0.6475869809203143

In [79]:
314/(314+577)

0.35241301907968575

### Knowledge found: 35.2% aboard the titanic where femalea and 64.8% where male

### 10.Female overall death rate?

In [80]:
sum([(df['Survived']==0) &(df['Sex']=='female')]).sum()

81

In [81]:
81/549

0.14754098360655737

### Knowledge found: 14.8% of the dead where Female

### 11. First class female survivability?

In [84]:
sum([(df['Survived']==0) & (df['Sex']=='female') & (df['Pclass']==1) ]).sum()

3

In [88]:
1-3/314

0.9904458598726115

### Knowledge found: First class Females had a 99%  of survivability